File: app/src/main/java/com/arbaelbarca/listmovieskatalog/network/ApiServices.java
package com.arbaelbarca.listmovieskatalog.network;

import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.ResponseDetailMovies;
import com.arbaelbarca.listmovieskatalog.model.modelreviewmovies.ResponseReviewMovies;
import com.arbaelbarca.listmovieskatalog.model.movies.ResponseMoviesList;

import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

public interface ApiServices {

    @GET("movie/{type}")
    Observable<ResponseMoviesList> getMovies(
            @Path("type") String type,
            @Query("language") String language
    );


    @GET("movie/{id_movie}")
    Observable<ResponseDetailMovies> getDetailMovies(
            @Path("id_movie") String type,
            @Query("language") String language
    );

    @GET("movie/{id_movie}/reviews")
    Observable<ResponseReviewMovies> getDetailReviews(
            @Path("id_movie") String type,
            @Query("language") String language
    );
}

File: app/src/main/java/com/arbaelbarca/listmovieskatalog/network/ApiServices.kt
package com.arbaelbarca.listmovieskatalog.network

import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.ResponseDetailMovies
import com.arbaelbarca.listmovieskatalog.model.modelreviewmovies.ResponseReviewMovies
import com.arbaelbarca.listmovieskatalog.model.movies.ResponseMoviesList

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import rx.Observable

interface ApiServices {

    @GET("movie/{type}")
    fun getMovies(
            @Path("type") type: String,
            @Query("language") language: String
    ): Observable<ResponseMoviesList>


    @GET("movie/{id_movie}")
    fun getDetailMovies(
            @Path("id_movie") type: String,
            @Query("language") language: String
    ): Observable<ResponseDetailMovies>

    @GET("movie/{id_movie}/reviews")
    fun getDetailReviews(
            @Path("id_movie") type: String,
            @Query("language") language: String
    ): Observable<ResponseReviewMovies>
}
